mephisto/data_model/task.py

```diff
@@ -362,6 +363,11 @@ def get_assignment_statuses(self) -> Dict[str, int]:
             for status in AssignmentState.valid()
         }
 
+    def update_completion_progress(self, task_launcher) -> None:
```

Comment:
Nit: should be used to both initialize `assignments_generator_done` to `False` as well as to set it `True` later. You can use the `task_launcher` to determine the state in both cases.